src/material/tests/material_properties_test.cc

```diff
@@ -0,0 +1,1537 @@
+  EXPECT_NO_THROW(MaterialProperties::CheckConsistent({{1, test_materials_.at(1)}}));
+  test_materials_.at(1).clear_matrix_property();
+  EXPECT_NO_THROW(MaterialProperties::CheckConsistent(test_materials_));
+  EXPECT_NO_THROW(MaterialProperties::CheckConsistent({{1, Material()}}));
+  test_materials_.at(1).set_number_of_groups(5);
+  test_materials_.at(2).set_number_of_groups(5);
+  EXPECT_NO_THROW(MaterialProperties::CheckConsistent({{1, test_materials_.at(1)}, {2, test_materials_.at(2)}}));
+  GetPropertyPtr(test_materials_.at(1), Material::ENERGY_GROUPS)->clear_value();
+  GetPropertyPtr(test_materials_.at(2), Material::ENERGY_GROUPS)->clear_value();
+  EXPECT_NO_THROW(MaterialProperties::CheckConsistent({{1, test_materials_.at(1)}, {2, test_materials_.at(2)}}));
+  test_materials_.at(1).clear_vector_property();
+  test_materials_.at(2).clear_vector_property();
+  EXPECT_NO_THROW(MaterialProperties::CheckConsistent({{1, test_materials_.at(1)}, {2, test_materials_.at(2)}}));
+}
+
+// tests for exceptions thrown from MaterialProperties Material map constructor
+
+TEST_F(MaterialPropertiesTest, NoFissileIDs) {
+  EXPECT_THROW(MaterialProperties mp(test_materials_, true, false, 7, 4), MaterialProperties::NoFissileIDs);
+  EXPECT_THROW(MaterialProperties mp(test_materials_, true, false, 7, 4, {}), MaterialProperties::NoFissileIDs);
+}
+
+TEST_F(MaterialPropertiesTest, FissileIDInvalid) {
+  EXPECT_THROW({
+    try {
+      MaterialProperties mp(test_materials_, true, false, 7, 4, {11, 5, 10});
+    }
+    catch (const MaterialProperties::FissileIDInvalid& e) {
+      std::string expected = "Material ID 5 was specified as fissile, but no material with ID 5 exists.";
+      EXPECT_EQ(expected, GetMessage(e));
+      throw;
+    }
+  }, MaterialProperties::FissileIDInvalid);
+
+  EXPECT_THROW({
+    try {
+      MaterialProperties mp(test_materials_, true, false, 7, 4, {-222});
+    }
+    catch (const MaterialProperties::FissileIDInvalid& e) {
+      std::string expected = "Material ID -222 was specified as fissile, but no material with ID -222 exists.";
+      EXPECT_EQ(expected, GetMessage(e));
+      throw;
+    }
+  }, MaterialProperties::FissileIDInvalid);
+
+  // test that the exception has one of the correct possible options
+  EXPECT_THROW({
+    try {
+      MaterialProperties mp(test_materials_, true, false, 7, 4, {1, 2, 3, 4, 8, 9, 10, 11});
+    }
+    catch (const MaterialProperties::FissileIDInvalid& e) {
+      std::string message = GetMessage(e);
+      bool errorMessageContainsInvalidID = false;
+      for (std::string id_str : {" 3 ", " 4 ", " 8 ", " 9 "})
+      {
+        if (message.substr(11, 3) == id_str) {
+          errorMessageContainsInvalidID = true;
+        }
+      }
+      EXPECT_TRUE(errorMessageContainsInvalidID);
+      throw;
+    }
+  }, MaterialProperties::FissileIDInvalid);
+}
+
+TEST_F(MaterialPropertiesTest, WrongNumberOfMaterials) {
+  EXPECT_THROW({
```

Comment:
The format is actually messy for all the `EXPECT_THROW` macros have `try` and `catch` in it. Remember, breaking a line in parentheses should follow one of the following two rules:
* all new lines should at least look up the open parenthesis (left)
* all new lines should based on indenting with 4 spaces.

Please change accordingly. Also pls read Google Style guide for this painful rule or discuss with me.